modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIPropertyRelevantObjectEntryModelListener.java

```diff
@@ -96,7 +103,36 @@ private void _scheduleAPIProperties(long apiPropertyId) {
 			});
 	}
 
-	private boolean _validate(
+	private void _validate(
+		long apiSchemaId, String objectFieldERC,
+		String objectRelationshipNames) {
+
+		try {
+			if (!_objectEntryHelper.isValidObjectEntry(
+					apiSchemaId, "L_API_SCHEMA")) {
+
+				throw new ObjectEntryValuesException.InvalidObjectField(
+					null, "An API property must be related to an API schema",
+					"an-api-property-must-be-related-to-an-api-schema");
+			}
+
+			if (!_validateAPIPropertyFields(
+					apiSchemaId, objectFieldERC, objectRelationshipNames)) {
+
+				throw new ObjectEntryValuesException.InvalidObjectField(
+					null,
+					"An API property must be related to an existing object " +
+						"field",
+					"an-api-property-must-be-related-to-an-existing-object-" +
+						"field");
+			}
+		}
+		catch (Exception exception) {
+			throw new ModelListenerException(exception);
+		}
+	}
+
+	private boolean _validateAPIPropertyFields(
```

Comment:
In the lines 135..144 we are performing the following request:

```
	private boolean _validateAPIPropertyFields(
			long apiSchemaId, String objectFieldExternalReferenceCode,
			String objectRelationshipName)
		throws Exception {

		ObjectEntry apiSchemaObjectEntry =
			_objectEntryLocalService.getObjectEntry(apiSchemaId);

		Map<String, Serializable> apiSchemaValues =
			apiSchemaObjectEntry.getValues();
```
however, we already had the ObjectEntry before the method invocation. Are we doing it properly? Maybe I am missunderstanding something :thinking:

Reply:
I think that in this case we are trying to get the related APISchema, the ObjectEntry that we had previously is the APIProperty, make sense?

Reply:
but for the create method, you will get nothing, right? And for the update method, you would have the previous one as an argument in the event thrown

Reply:
I got confused, it is ok how it is